### PR TITLE
[OV20] NV12toRGB and NV12toBGR operations specification

### PR DESCRIPTION
**Pull request by nosovmik**
### Details:
 - NV12toRGB and NV12toBGR operations specification
 - It was decided to split this into 2 separate operations to be able to maintain their behavior separately. Also in futurethere may be needed to perform conversion from NV21, and this extension shall not affect users which use old operations (e.g. NV12toRGB-8 will be the same even if there is new NV21toRGB operation is introduced)

### Tickets:
 - 62174

### CODE REVIEW

**Review by ilya-lavrenov — APPROVED** [review]

File: docs/ops/image/NV12toBGR_8.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+* **1**: A tensor of type *T* representing converted image in BGR format. Dimensions:
+  * `N` - batch dimension
+  * `H` - height dimension is same as image height
+  * `W` - width dimension is same as image width
+  * `C` - channels dimension equals to 3. First channel is Blue, second one is Green, last one is Red
+
+**Types:**
+
+* *T*: `uint8` or `float32` type.
```
Comment (vladimir-paramuzov) [inline docs/ops/image/NV12toBGR_8.md:27]: Shouldn't `float32` be replaced with `any supported floating-point type`?

Comment (nosovmik) [reply docs/ops/image/NV12toBGR_8.md:27]: @ilya-lavrenov can you comment on this? Technically any type except 'bool', 'int4', 'int8' is fine to store range 0-255

Comment (ilya-lavrenov) [reply docs/ops/image/NV12toBGR_8.md:27]: I'm OK with it, but will plugins support it?

@jane-intel what is the policy for such cases?

File: docs/doxygen/ie_docs.xml [hunk 2]
```diff
@@ -211,6 +211,8 @@ limitations under the License.
                 <tab type="user" title="NonZero-3" url="@ref openvino_docs_ops_condition_NonZero_3"/>
                 <tab type="user" title="NormalizeL2-1" url="@ref openvino_docs_ops_normalization_NormalizeL2_1"/>
                 <tab type="user" title="NotEqual-1" url="@ref openvino_docs_ops_comparison_NotEqual_1"/>
+                <tab type="user" title="NV12toBGR-8" url="@ref openvino_docs_ops_image_NV12toBGR_8"/>
+                <tab type="user" title="NV12toRGB-8" url="@ref openvino_docs_ops_image_NV12toRGB_8"/>
```
Comment (vinograd47) [inline docs/doxygen/ie_docs.xml:215]: Just curios. Will we have `BGRtoRGB` operation or can it be already represented in nGraph? If yes, the `NV12toRGB-8` can be represented as a sequence of `NV12toBGR-8 -> BGRtoRGB` and allows to implementation to fuse them if supported. IMHO, this is more nGraph way.

Comment (nosovmik) [reply docs/doxygen/ie_docs.xml:215]: Good question, we're going to have RGB<->BGR conversion in future. @jane-intel, can you please share your thoughts?

Comment (jane-intel) [reply docs/doxygen/ie_docs.xml:215]: 1. We won't have RGB->BGR, we already have it through Gather operation or Split->Concat if you wish. No need to extend opset with those.
2. Leaving one operation `NV12toBGR` will make us provide a transformation to fuse `NV12toBGR->Gather` to `SuperPrivateNV12toRGB` for all the plugins, so we 1 -- won't reduce binary size and 2 -- will extend load time with yet another transformation.

File: docs/ops/image/NV12toBGR_8.md (hunk 1 above)
Comment (jane-intel) [reply docs/ops/image/NV12toBGR_8.md:27]: *T*: `uint8` or any supported floating-point type.

**Review by ilyachur — APPROVED** [review]

File: docs/ops/image/NV12toBGR_8.md (hunk 1 above)
Comment (nosovmik) [reply docs/ops/image/NV12toBGR_8.md]: Thanks, corrected

**Review by vladimir-paramuzov — APPROVED** [review]

**Comment by nosovmik** [comment]
@andrew-zaytsev can you please review this spec from your side?

**Review by ArtemySkrebkov-intel — APPROVED** [review]

File: docs/ops/image/NV12toBGR_8.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+Same as specified for [NV12toRGB](NV12toRGB_8.md) operation.
+
+**Outputs:**
+
+* **1**: A tensor of type *T* representing converted image in BGR format. Dimensions:
+  * `N` - batch dimension
+  * `H` - height dimension is same as image height
+  * `W` - width dimension is same as image width
+  * `C` - channels dimension equals to 3. First channel is Blue, second one is Green, last one is Red
```
Comment (tsavina) [inline docs/ops/image/NV12toBGR_8.md:23]:
```suggestion
  * `C` - channels dimension is equal to 3. The first channel is Blue, the second one is Green, the last one is Red
```

File: docs/ops/image/NV12toBGR_8.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+**Short description**: *NV12toBGR* performs image conversion from NV12 format to BGR.
+
+**Detailed description**:
+
+Similar to *NV12toRGB* but output channels for each pixel are reversed so that: first channel is `blue`, second one is `green`, last one is `red`.  See detailed conversion formulas at [NV12toRGB](NV12toRGB_8.md)
+
+**Inputs:**
+
+Same as specified for [NV12toRGB](NV12toRGB_8.md) operation.
```
Comment (jane-intel) [inline docs/ops/image/NV12toBGR_8.md:15]: does this link work?

File: docs/ops/image/NV12toBGR_8.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+**Outputs:**
+
+* **1**: A tensor of type *T* representing converted image in BGR format. Dimensions:
+  * `N` - batch dimension
+  * `H` - height dimension is same as image height
```
Comment (tsavina) [inline docs/ops/image/NV12toBGR_8.md:21]:
```suggestion
  * `H` - height dimension is the same as the image height
```

File: docs/ops/image/NV12toBGR_8.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+  * `W` - width dimension is same as image width
```
Comment (tsavina) [inline docs/ops/image/NV12toBGR_8.md:22]:
```suggestion
  * `W` - width dimension is the same as the image width
```

File: docs/ops/image/NV12toBGR_8.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+Similar to *NV12toRGB* but output channels for each pixel are reversed so that: first channel is `blue`, second one is `green`, last one is `red`.  See detailed conversion formulas at [NV12toRGB](NV12toRGB_8.md)
+
+**Inputs:**
+
+Same as specified for [NV12toRGB](NV12toRGB_8.md) operation.
+
+**Outputs:**
+
+* **1**: A tensor of type *T* representing converted image in BGR format. Dimensions:
```
Comment (tsavina) [inline docs/ops/image/NV12toBGR_8.md:19]:
```suggestion
* **1**: A tensor of type *T* representing an image converted in BGR format. Dimensions:
```

File: docs/ops/image/NV12toBGR_8.md (hunk 4 above)
Comment (nosovmik) [reply docs/ops/image/NV12toBGR_8.md:15]: I tried it locally, it worked well. In Azure it seems that upload of generated docs is disabled
image

File: docs/ops/image/NV12toBGR_8.md [hunk 8]
```diff
@@ -0,0 +1,82 @@
+## NV12toBGR <a name="NV12toBGR"></a> {#openvino_docs_ops_image_NV12toBGR_8}
+
+**Versioned name**: *NV12toBGR-8*
+
+**Category**: *Image processing*
+
+**Short description**: *NV12toBGR* performs image conversion from NV12 format to BGR.
```
Comment (tsavina) [inline docs/ops/image/NV12toBGR_8.md:7]:
```suggestion
**Short description**: *NV12toBGR* performs image conversion from NV12 to BGR format.
```

File: docs/ops/image/NV12toRGB_8.md [hunk 9]
```diff
@@ -0,0 +1,109 @@
+## NV12toRGB <a name="NV12toRGB"></a> {#openvino_docs_ops_image_NV12toRGB_8}
+
+**Versioned name**: *NV12toRGB-8*
+
+**Category**: *Image processing*
+
+**Short description**: *NV12toRGB* performs image conversion from NV12 format to RGB format.
```
Comment (tsavina) [inline docs/ops/image/NV12toRGB_8.md:7]:
```suggestion
**Short description**: *NV12toRGB* performs image conversion from NV12 to RGB format.
```

File: docs/ops/image/NV12toRGB_8.md [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+
+**Detailed description:**
+
+Conversion of each pixel from NV12 (YUV) to RGB space is represented by following formulas:
```
Comment (tsavina) [inline docs/ops/image/NV12toRGB_8.md:11]:
```suggestion
Conversion of each pixel from NV12 (YUV) to RGB space is represented by the following formulas:
```

File: docs/ops/image/NV12toRGB_8.md [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+
+\f[
+\begin{aligned}
+& R = 1.164 \cdot (Y - 16) + 1.596 \cdot (V - 128) \\
+& G = 1.164 \cdot (Y - 16) - 0.813 \cdot (V - 128) - 0.391 \cdot (U - 128) \\
+& B = 1.164 \cdot (Y - 16) + 2.018 \cdot (U - 128)
+\end{aligned}
+\f]
+
+Then R, G, B values are clipped to range (0, 255)
```
Comment (tsavina) [inline docs/ops/image/NV12toRGB_8.md:21]:
```suggestion
Then R, G, B values are clipped to range (0, 255).
```

File: docs/ops/image/NV12toRGB_8.md [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+
+**Inputs:**
+
+Input NV12 image tensor shall have `NHWC (a.k.a NYXC)` layout and can be represented in two ways:
```
Comment (tsavina) [inline docs/ops/image/NV12toRGB_8.md:25]:
```suggestion
Input NV12 image tensor shall have `NHWC (also known as NYXC)` layout and can be represented in two ways:
```

File: docs/ops/image/NV12toRGB_8.md [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+* *Single plane*:
+  * **1**: Tensor of type *T*. **Required.** Dimensions:
+    * `N` - batch dimension
+    * `H` - height dimension is 1.5x bigger than image height
```
Comment (tsavina) [inline docs/ops/image/NV12toRGB_8.md:29]:
```suggestion
    * `H` - height dimension is 1.5x bigger than the image height
```

File: docs/ops/image/NV12toRGB_8.md [hunk 14]
```diff
@@ -0,0 +1,109 @@
...
+    * `W` - width dimension is same as image width
```
Comment (tsavina) [inline docs/ops/image/NV12toRGB_8.md:30]:
```suggestion
    * `W` - width dimension is the same as the image width
```

File: docs/ops/image/NV12toRGB_8.md [hunk 15]
```diff
@@ -0,0 +1,109 @@
...
+    * `C` - channels dimension equals to 1 (one plane)
```
Comment (tsavina) [inline docs/ops/image/NV12toRGB_8.md:31]:
```suggestion
    * `C` - channels dimension is equal to 1 (one plane)
```

File: docs/ops/image/NV12toRGB_8.md [hunk 16]
```diff
@@ -0,0 +1,109 @@
...
+* *Two separate planes: Y and UV*: In this way
+  * **1**: Tensor of type *T* representing Y plane. **Required.** Dimensions:
+    * `N` - batch dimension
+    * `H` - height dimension is same as image height
```
Comment (tsavina) [inline docs/ops/image/NV12toRGB_8.md:35]:
```suggestion
    * `H` - height dimension is the same as the image height
```

File: docs/ops/image/NV12toRGB_8.md [hunk 17]
```diff
@@ -0,0 +1,109 @@
...
+    * `W` - width dimension is same as image width
```
Comment (tsavina) [inline docs/ops/image/NV12toRGB_8.md:36]:
```suggestion
    * `W` - width dimension is the same as the image width
```

File: docs/ops/image/NV12toRGB_8.md [hunk 18]
```diff
@@ -0,0 +1,109 @@
...
+    * `C` - channels dimension equals to 1 (only Y channel)
```
Comment (tsavina) [inline docs/ops/image/NV12toRGB_8.md:37]:
```suggestion
    * `C` - channels dimension is equal to 1 (only Y channel)
```

File: docs/ops/image/NV12toRGB_8.md [hunk 19]
```diff
@@ -0,0 +1,109 @@
...
+  * **2**: Tensor of type *T* representing UV plane. **Required.** Dimensions:
+    * `N` - batch dimension. Shall be same as batch dimension for Y plane
```
Comment (tsavina) [inline docs/ops/image/NV12toRGB_8.md:39]:
```suggestion
    * `N` - batch dimension. Shall be the same as the batch dimension for Y plane
```

File: docs/ops/image/NV12toRGB_8.md [hunk 20]
```diff
@@ -0,0 +1,109 @@
...
+    * `H` - height dimension shall be half of image height (i.e. `image_height / 2`)
```
Comment (tsavina) [inline docs/ops/image/NV12toRGB_8.md:40]:
```suggestion
    * `H` - height dimension shall be half of the image height (for example, `image_height / 2`)
```

File: docs/ops/image/NV12toRGB_8.md [hunk 21]
```diff
@@ -0,0 +1,109 @@
...
+    * `W` - width dimension shall be half of image width (i.e. `image_width / 2`)
```
Comment (tsavina) [inline docs/ops/image/NV12toRGB_8.md:41]:
```suggestion
    * `W` - width dimension shall be half of the image width (for example, `image_width / 2`)
```

File: docs/ops/image/NV12toRGB_8.md [hunk 22]
```diff
@@ -0,0 +1,109 @@
...
+    * `C` - channels dimension shall be equal to 2 (U channel and V channel)
+
+**Outputs:**
+
+* **1**: A tensor of type *T* representing converted image in RGB format. Dimensions:
```
Comment (tsavina) [inline docs/ops/image/NV12toRGB_8.md:46]:
```suggestion
* **1**: A tensor of type *T* representing an image converted in RGB format. Dimensions:
```

File: docs/ops/image/NV12toRGB_8.md [hunk 23]
```diff
@@ -0,0 +1,109 @@
...
+  * `N` - batch dimension
+  * `H` - height dimension is same as image height
```
Comment (tsavina) [inline docs/ops/image/NV12toRGB_8.md:48]:
```suggestion
  * `H` - height dimension is the same as the image height
```

File: docs/ops/image/NV12toRGB_8.md [hunk 24]
```diff
@@ -0,0 +1,109 @@
...
+  * `W` - width dimension is same as image width
```
Comment (tsavina) [inline docs/ops/image/NV12toRGB_8.md:49]:
```suggestion
  * `W` - width dimension is the same as the image width
```

File: docs/ops/image/NV12toRGB_8.md [hunk 25]
```diff
@@ -0,0 +1,109 @@
...
+  * `C` - channels dimension equals to 3. First channel is Red, second one is Green, last one is Blue
```
Comment (tsavina) [inline docs/ops/image/NV12toRGB_8.md:50]:
```suggestion
  * `C` - channels dimension is equal to 3. The first channel is Red, the second one is Green, the last one is Blue
```

File: docs/ops/image/NV12toRGB_8.md [hunk 26]
```diff
@@ -0,0 +1,109 @@
...
+**Inputs:**
+
+Input NV12 image tensor shall have `NHWC (a.k.a NYXC)` layout and can be represented in two ways:
+* *Single plane*:
+  * **1**: Tensor of type *T*. **Required.** Dimensions:
+    * `N` - batch dimension
+    * `H` - height dimension is 1.5x bigger than image height
+    * `W` - width dimension is same as image width
+    * `C` - channels dimension equals to 1 (one plane)
+* *Two separate planes: Y and UV*: In this way
```
Comment (tsavina) [inline docs/ops/image/NV12toRGB_8.md:32]:
```suggestion
* *Two separate planes - Y and UV*:
```

**Review by tsavina — APPROVED** [review]

**Comment by nosovmik** [comment]
@vinograd47 @dmitry-gorokhov can you please approve specification from CPU plugin side?

**Review by vinograd47 — APPROVED** [review]

**Review by jane-intel — APPROVED** [review]

File: docs/doxygen/ie_docs.xml (hunk 2 above)
Comment (ilya-lavrenov) [reply docs/doxygen/ie_docs.xml:215]: In this case should we add `NV12toRGBp` to have fused verison of `NV12toRGB` and `transpose`?

**Pull request merged**